### PR TITLE
Restructure and expand Project Name documentation

### PR DESCRIPTION
**Pull request by smokris**
- Split into a section for each type of project, for improved clarity
- Add syntax for Multibranch Pipeline projects (as mentioned on https://issues.jenkins-ci.org/browse/JENKINS-40429#comment-343463)

### CODE REVIEW

**Comment by smokris** [comment]
Oh, I didn't realize this was a global behavior; in that case I think a wiki page makes sense.  I reworked this documentation into a new wiki page: https://wiki.jenkins.io/display/JENKINS/How+to+reference+another+project+by+name.

File: src/main/resources/hudson/plugins/copyartifact/CopyArtifact/help-projectName.html [hunk 1]
```diff
@@ -1,20 +1,39 @@
...
+      To copy artifacts from one matrix project to another, use a parameter to
+      select the matching configuration in the source project.<br />
+      <strong>Example:</strong> <tt>OtherMatrixJob/jdk=$jdk</tt>
+    </p>
+  </dd>
+  <dt><strong>Multibranch Pipeline projects:</strong></dt>
+  <dd>
+    <p>
+      Use an absolute path consisting of the project name followed by the branch name.<br />
+      <strong>Example:</strong> <tt>/MyMultibranchProject/MyBranch</tt>
```
[the remark this answers is not in the document]
Comment (ikedam) [reply src/main/resources/hudson/plugins/copyartifact/CopyArtifact/help-projectName.html]: Added in 72123fa69f84afc6ba8b79fe7c6caf70e13b8536 .

File: src/main/resources/hudson/plugins/copyartifact/CopyArtifact/help-projectName.html [hunk 2]
```diff
@@ -1,20 +1,39 @@
...
+    <p>
+      To copy artifacts from one matrix project to another, use a parameter to
+      select the matching configuration in the source project.<br />
+      <strong>Example:</strong> <tt>OtherMatrixJob/jdk=$jdk</tt>
+    </p>
+  </dd>
+  <dt><strong>Multibranch Pipeline projects:</strong></dt>
+  <dd>
+    <p>
+      Use an absolute path consisting of the project name followed by the branch name.<br />
```
[the remark this answers is not in the document]
Comment (ikedam) [reply src/main/resources/hudson/plugins/copyartifact/CopyArtifact/help-projectName.html]: I tested and verified that this need not to be an abosolute path.
Updated the text in 72123fa69f84afc6ba8b79fe7c6caf70e13b8536.

File: src/main/resources/hudson/plugins/copyartifact/CopyArtifact/help-projectName.html [hunk 3]
```diff
@@ -1,20 +1,45 @@
...
+    <p>
+      To copy artifacts from one matrix project to another, use a parameter to
+      select the matching configuration in the source project.<br />
+      <strong>Example:</strong> <tt>OtherMatrixJob/jdk=$jdk</tt>
+    </p>
+  </dd>
+  <dt><strong>Multibranch Pipeline projects:</strong></dt>
+  <dd>
+    <p>
+      Use an path consisting of the project name followed by the branch name.<br />
```
Comment (smokris) [inline src/main/resources/hudson/plugins/copyartifact/CopyArtifact/help-projectName.html]:
```suggestion
      Use a path consisting of the project name followed by the branch name.<br />
```

Comment (ikedam) [reply src/main/resources/hudson/plugins/copyartifact/CopyArtifact/help-projectName.html]: Thanks! I fixed that.

**Comment by ikedam** [comment]
Released copyartifact-1.42.1 with this fix. It will be available in the update center in a day.
Thanks for the improvement.

**Comment by ikedam** [comment]
https://issues.jenkins-ci.org/browse/JENKINS-39174

**Pull request merged**